File: Data Structures & Algorithms/eating-bananas/submission-6.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        max_value = 0

        for pile in piles:
            max_value = max(max_value, pile)
        
        left = 1
        right = max_value

        while left <= right:

            middle = (left + right) // 2
            sum = 0

            for pile in piles:
                sum += math.ceil((pile / middle))
            
            print(sum)

            if sum <= h:
                right = middle - 1
            else:
                left = middle + 1
        
        return left

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-6.java
class Solution {
    public int findMin(int[] nums) {
        int left = 0;
        int right = nums.length - 1;
        int minValue = Integer.MAX_VALUE;

        while(left <= right){
            int middle = (left + right) / 2;

            if(nums[left] < nums[right] && nums[left] < minValue) return nums[left];

            minValue = Math.min(minValue, nums[middle]);

            if(nums[left] <= nums[middle]){
                left = middle + 1;
            }else {
                right = middle - 1;
            }
        }

        return minValue;
    }
}
